internal/planning/invocation/invocation.go

```diff
@@ -33,11 +32,8 @@ type Invocation struct {
 	Image                compute.Computable[oci.ResolvableImage]
 	PublicImageID        *oci.ImageID
 	SupportedToolVersion int
-	Command              []string
-	Args                 []string
-	Env                  []*schema.BinaryConfig_EnvEntry
+	BinaryConfig         *schema.BinaryConfig
```

Comment:
This feels like a step backwards because we're now embedding a single definition into domain-specific invocations. Previously, if one would add a new field to BinaryConfig, it would be clear whether there's a place for it in e.g. `Invocation` or not -- now there's an assumption that BinaryConfig is universally applicable, and that doesn't feel future-proof.